enclave/src/polkadex_nonce_storage/nonce_storage.rs

```diff
@@ -0,0 +1,68 @@
+// This file is part of Polkadex.
+
+// Copyright (C) 2020-2021 Polkadex oü and Supercomputing Systems AG
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use polkadex_sgx_primitives::AccountId;
+use log::*;
+use codec::Encode;
+use sgx_tstd::collections::HashMap;
+use sgx_tstd::vec::Vec;
+
+pub type EncodedKey = Vec<u8>;
+
+#[derive(Debug)]
+pub struct PolkadexNonceStorage {
+    /// map AccountId -> NonceHandler
+    pub storage: HashMap<EncodedKey, u32>,
+}
+
+impl PolkadexNonceStorage {
+    pub fn create() -> PolkadexNonceStorage {
+        PolkadexNonceStorage {
+            storage: HashMap::new(),
+        }
+    }
+
+    pub fn read_nonce(&mut self, acc: AccountId) -> u32 {
+        debug!("reading nonce from acc: {:?}", acc);
+        if let Some(nonce) = self.storage.get(&acc.clone().encode()) {
+            *nonce
+        }
+        else {
+            self.initialize_nonce(acc);
```

Comment:
I agree as long as we are not storing the nonce on-chain. This should be something local to the enclave, and a user will have a unique nonce for each trading pair since there will be a new worker instance for each trading pair.

Also, this approach for nonce can create problems when we load balancing using multiple instances of workers, but that can be looked into later. 

For now, let's proceed with @haerdib's approach

Reply:
Ok, I will proceed with initialization on account register and will change that function to return an error in case it's trying to get an uninitialized nonce.